packages/components/src/internal/app/utils.ts

```diff
@@ -124,6 +124,10 @@ export function isSampleManagerEnabled(): boolean {
     return getServerContext().moduleContext?.samplemanagement !== undefined;
 }
 
+export function hasPremiumModule(): boolean {
+    return getServerContext().moduleContext?.samplemanagement?.hasPremiumModule ?? false;
```

Comment:
I was just piggybacking on the existing SM module context which includes the flag for `hasStudyModule`. Also, it looks like the module context for any given module is only included on the LABKEY object for enabled modules. So the main scenario here is for the Apps like SM and LKB to be able to tell if the premium module is present at all instead of needing to know if it is enabled.